okhttp-clients/src/main/java/com/palantir/conjure/java/okhttp/HostEventsSink.java

```diff
@@ -41,10 +46,55 @@ public void recordIoException() {
         };
     }
 
-    interface HostEventCallback {
+    /**
+     * @deprecated prefer super interface
+     */
+    @Deprecated
+    interface HostEventCallback extends com.palantir.conjure.java.client.config.HostEventsSink.HostEventCallback {
 
+        @Override
         void record(int statusCode, long micros);
 
+        @Override
         void recordIoException();
+
+        static HostEventCallback from(com.palantir.conjure.java.client.config.HostEventsSink.HostEventCallback other) {
+            if (other instanceof HostEventCallback) {
+                return (HostEventCallback) other;
+            }
+            return new HostEventCallback() {
+                @Override
+                public void record(int statusCode, long micros) {
+                    other.record(statusCode, micros);
+                }
+
+                @Override
+                public void recordIoException() {
+                    other.recordIoException();
+                }
+            };
+        }
+    }
+
+    static HostEventsSink from(com.palantir.conjure.java.client.config.HostEventsSink other) {
```

Comment:
I had to add all these gross converter things because OkHttpClients.java has methods which revapi otherwise says would ABI break.

```
public static OkHttpClient create(
            OkHttpClient.Builder client,
            ClientConfiguration config,
            UserAgent userAgent,
            HostEventsSink hostEventsSink,
            Class<?> serviceClass) {
```

Reply:
I like it, thanks!